app/models/payment_confirmation.rb

```diff
@@ -0,0 +1,87 @@
+class PaymentConfirmation
+  attr_reader :payroll_run, :csv_file, :updated_claim_ids
+  attr_accessor :errors
+
+  def initialize(payroll_run, csv_file)
+    @payroll_run = payroll_run
+    @errors = []
+    @csv_file = csv_file
+    @line_number = 1
+    @updated_claim_ids = Set.new
+  end
+
+  def ingest
+    return unless valid?
+
+    ActiveRecord::Base.transaction do
+      csv.body.each do |row|
+        @line_number += 1
+        payment = fetch_payment_by_reference(row["Claim ID"])
+        update_payment_fields(payment, row) unless payment.nil?
+      end
+
+      valid? || (raise ActiveRecord::Rollback)
```

Comment:
It's the way it is because we want the method to return something truthy on success; it would return `nil` with the suggested change.

Reply:
Ah right, that makes sense. It's a shame the standard Ruby gem complains about the use of the `and` and `or` operators as this would be a perfect place to use `or`!